impeller/renderer/backend/vulkan/pipeline_cache_vk.h

```diff
@@ -8,14 +8,16 @@
 #include "flutter/fml/macros.h"
 #include "impeller/base/thread.h"
 #include "impeller/renderer/backend/vulkan/capabilities_vk.h"
+#include "impeller/renderer/backend/vulkan/device_holder.h"
 #include "impeller/renderer/backend/vulkan/vk.h"
 
 namespace impeller {
 
 class PipelineCacheVK {
  public:
   explicit PipelineCacheVK(std::shared_ptr<const Capabilities> caps,
-                           vk::Device device,
+                           std::weak_ptr<DeviceHolder> device_holder,
+                           const vk::Device* device,
```

Comment:
Remove the `device` - it's redundant with the `device_holder`

Reply:
It's not redundant unfortunately.  The way we initialize the `ContextVK` the `DeviceHolder` is not valid at the time we call `PipelineCacheVK()`.  It's because in the ContextVK constructor all data is held on the stack and only if everything is successful are the ivar's written to and `is_valid_` is set to true.

Our options were:
1) Send in the Device and avoid calling into the DeviceHolder
1) Stop using the Device in the PipelineCacheVK constructor
1) Write the Device to the ivar of ContextVK but make a RAII wrapper that unsets it if we don't get to `is_valid_ = true`.

I went for the first option.  I can add a comment to make it more clear.  The second option is probably a good design but is a risky change (I'd rather do that as a followup if we wanted to).  The 3rd option seemed to have just as much baggage as "don't use the DeviceHolder if you have the Device".

Reply:
A comment SGTM.  Later we can look into ways to make construction of `PipelineCacheVK` cleaner.